app/src/components/App/Application/index.jsx

```diff
@@ -52,65 +53,30 @@ class Application extends Component {
     });
   }
 
-  async submitApplication() {
+  submitApplication() {
     const { name, email,
-      applicationText: application_text,
-      ordered: prioritized,
-      selectedComittees: committees,
+      applicationText,
+      ordered,
+      selectedComittees,
     } = this.state;
-    const application = {
-      name, email, prioritized, application_text,
-      committees: committees.map((committee, index) => Object.assign({
-        group: committeesMap.get(committee).id,
-        priority: index + 1,
-      })),
-    };
-
-    const uri = `${process.env.SG_APPLICATION_BACKEND}${process.env.SG_APPLICATION_ENDPOINT}`;
+  
+    const application = new CommitteeApplication(null, name, email, applicationText, selectedComittees, ordered);
+    
     this.setState({ disableSubmit: true });
-
-    const headers = {
-      'Content-Type': 'application/json'
-    }
-    if (this.props.user && this.props.user._access_token.length > 0) {
-      headers.Authorization = `Bearer ${this.props.user._access_token}`;
-      application.name = '';
-      application.email = '';
-    }
-
-    try {
-      const resp = await fetch(uri, {
-        headers,
-        body: JSON.stringify(application),
-        method: 'POST',
+    this.props.applicationService.postApplication(application).subscribe(() => {
+      //Everything went ok
+      this.setState({
+        responseMessage: 'Søknad er sendt.',
       });
-
-      const json = await resp.json();
-
-      if (resp.status === 201) {
-        this.setState({
-          responseMessage: 'Søknad er sendt.',
-        });
-        return;
-      }
-
-      // Sometimes we get an array of messages and sometimes we get
-      // an object. If we get an array we can concat it together,
-      // if we get an object we show a more generic error message.
-      const errorMessage = json instanceof Array ? json : 'Pass på at du har fylt ut alle feltene og valgt komiteer å søke.';
-
+    },(err) => {
```

Comment:
It is based on the respon's `ok` param, if it isn't ok an error is thrown. That means all 4XX errors will be cough.